Add safeguards to user self-removal feature

MrPetovan opened the pull request:
Fix #13311 

Thanks to @annando for these suggestions:
- Add exception when authentication returns different user id than logged in in Settings\RemoveMe
- Throw exception when user has delegates in User::remove
- Add current user's hovercard to removeme page

New look:
![image](https://github.com/friendica/friendica/assets/925415/e7b52903-1393-4c77-b429-b1b455999438)

MrPetovan committed:
Add current user's hovercard to removeme page

- Extract Hovercard generation to Content\Widget

MrPetovan committed:
Throw exception when user has delegates in User::remove

MrPetovan committed:
Add exception when authentication returns different user id than logged in in Settings\RemoveMe

annando commented:
Seems like you have to rebuild the `messages.po`.

MrPetovan committed:
Updated main translation file after adding strings

annando commented on src/Model/User.php:
```diff
@@ -1620,16 +1620,24 @@ public static function sendRegisterOpenEmail(L10n $l10n, array $user, string $si
+			throw new HTTPException\NotFoundException('User not found with uid: ' . $uid);
+		}
+
+		if (DBA::exists('user', ['parent-uid' => $uid])) {
```
There is an additional way to delegate, using the `manage` table. 

P.S.: I just detected the function `User::hasIdentities($uid)`.

MrPetovan replied:
`User::getIdentities` returns results even for delegated accounts that can be freely deleted, making it unsuitable for this check.

I’ll add an exception for manager accounts, although I’m not sure what’s the way out, can manager accounts sever the management link?

annando replied:
From my point of view we should respect all of this.

MrPetovan replied:
That's not my question. When the User::remove call fails because the account is recorded as a manager for another account, can they sever the management link themselves? Should this check only be done during self-removal? What about when an account is removed from the moderation section?

annando replied:
I must confess that for a long term I hadn't got a look at this kind of delegation. We should test this out.

AlfredSK replied:
Are you talking about the possibility to manage other accounts via the 'delegate' way? As a delegate I am not able to access the settings of a delagated account (403 error) and thus the 'remove account' dialogue isn't accessible as well.

annando replied:
I know. But I had the idea that you mostly don't want to remove an account that is used to manage other accounts. And if you want to, you have to remove the access.

MrPetovan committed:
Refactor Delegation modules

- Remove dependency on DI
- Group translation labels in template variables
- Reformat tempate code

MrPetovan commented:
Blocked by #13315

annando commented:
I think it would be great, having this functionality in the next version. Do you think, that you can finalize it, or at least create some intermediate functionality? I think that even the current state is a real improvement.

MrPetovan commented:
I will not be able to rework the whole delegation feature in time, I guess I can bring it up to the current state.

MrPetovan commented:
The messages.po check is failing for a couple of instances of different line wraps, probably because I'm running the script on my Windows machine.

annando commented:
What is the current state here? I really would like to have it in the upcoming release.

MrPetovan commented:
Let me rebase this branch.

MrPetovan commented:
Rebase done.

tobiasd commented:
In case this PR shall be part of the 2023.09 (10) release, please redirect it towards the 2023.09-rc branch.

MrPetovan commented:
Thank you for the heads-up!

annando commented:
Sadly just another conflict ...

MrPetovan commented:
On it.

MrPetovan commented:
Done.

Some tests now fail:
```
There were 3 failures:

1) Friendica\Test\src\Content\SmiliesTest::testIsEmojiPost with data set "zwj" (true, '👨‍👨‍👧‍')
Failed asserting that false matches expected true.

/woodpecker/src/github.com/friendica/friendica/tests/src/Content/SmiliesTest.php:144

2) Friendica\Test\src\Content\SmiliesTest::testIsEmojiPost with data set "zwj+whitespace" (true, ' 👨‍👨‍👧‍ ')
Failed asserting that false matches expected true.

/woodpecker/src/github.com/friendica/friendica/tests/src/Content/SmiliesTest.php:144

3) Friendica\Test\src\Content\SmiliesTest::testIsEmojiPost with data set "zwj+HTML whitespace" (true, '&nbsp;👨‍👨‍👧‍&nbsp;')
Failed asserting that false matches expected true.

/woodpecker/src/github.com/friendica/friendica/tests/src/Content/SmiliesTest.php:144
```
Locally, they pass though, so I assume it's because the `php-intl` module isn't enabled in the CI image as the zero-width joiner (zwj) tests are meant to challenge the assumption that emoji characters are all coded on 4 bytes, which is the fallback if the `intl` module isn't enabled.

annando commented:
Yeah. That one fail because the fall back mechanism doesn't handle this emoji correctly.
